### PR TITLE
network: splitting Firewalls out into their own package

### PR DESCRIPTION
**Pull request by tombuildsstuff**
This is part of some larger work in the networking package to make this more maintainable and limits the blast radius of Networking SDK upgrades

This PR's mostly just a lift & shift - nothing particularly exciting to note

Dependent on #9772

### CODE REVIEW

**Review by jackofallops — APPROVED** [review]
Thanks @tombuildsstuff - just a couple super minor comments, but otherwise LGTM 👍

File: azurerm/internal/services/firewall/tests/firewall_policy_rule_collection_group_resource_test.go [hunk 1]
```diff
@@ -8,7 +8,7 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/terraform"
 	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/acceptance"
 	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
-	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/network/parse"
+	parse2 "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/firewall/parse"
```
[the remark this answers is not in the document]
Comment (tombuildsstuff) [reply azurerm/internal/services/firewall/tests/firewall_policy_rule_collection_group_resource_test.go:11]: done

File: azurerm/internal/services/firewall/tests/firewall_policy_resource_test.go [hunk 2]
```diff
@@ -8,7 +8,7 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/terraform"
 	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/acceptance"
 	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
-	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/network/parse"
+	parse2 "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/firewall/parse"
```
[the remark this answers is not in the document]
Comment (tombuildsstuff) [reply azurerm/internal/services/firewall/tests/firewall_policy_resource_test.go:11]: done

File: azurerm/internal/services/firewall/firewall_resource.go [hunk 3]
```diff
@@ -14,7 +13,8 @@ import (
 	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
 	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
 	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/locks"
-	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/network/validate"
+	firewallValidate "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/firewall/validate"
```
[the remark this answers is not in the document]
Comment (tombuildsstuff) [reply azurerm/internal/services/firewall/firewall_resource.go:16]: done

**Pull request merged**